great_expectations/data_context/util.py

```diff
@@ -462,7 +462,16 @@ def substitute_all_config_variables(
         data = DataContextConfigSchema().dump(data)
 
     if isinstance(data, CheckpointConfig):
-        data = CheckpointConfigSchema().dump(data)
+        if (
```

Comment:
@NathanFarmer Same question...  Is `data.batch_request` at this level -- configuration object -- ever *not* a `dict`?..

Reply:
@alexsherstinsky An in-memory DataFrame must be passed into a `RuntimeBatchRequest` object which in turn is passed into a `Checkpoint` at run time. The `CheckpointConfig` is instantiated in the `Checkpoint.__init__` and not ever converted to `dict`.

Reply:
@alexsherstinsky See the latest changes which convert `BatchRequest` objects to `dict` at `CheckpointConfig` instantiation.